lib/functions/general/git.sh

```diff
@@ -275,21 +275,11 @@ function fetch_from_repo() {
 				display_alert "Skipping submodules" "GIT_SKIP_SUBMODULES=yes" "debug"
 			else
 				display_alert "Updating submodules" "" "ext"
-				# FML: http://stackoverflow.com/a/17692710
-				for i in $(git config -f .gitmodules --get-regexp path | awk '{ print $2 }'); do
-					cd "${git_work_dir}" || exit
-					local surl sref
-					surl=$(git config -f .gitmodules --get "submodule.$i.url")
-					sref=$(git config -f .gitmodules --get "submodule.$i.branch" || true)
-					if [[ -n $sref ]]; then
-						sref="branch:$sref"
-					else
-						sref="head"
-					fi
-					display_alert "Updating submodule" "$i - $surl - $sref" "git"
-					git_ensure_safe_directory "$workdir/$i"
-					fetch_from_repo "$surl" "$workdir/$i" "$sref"
-				done
+				git submodule update --init --recursive --checkout
```

Comment:
Before: if no submodules, nothing would run.
Now: always runs.

Reply:
What about doing it this way? https://github.com/armbian/build/pull/8962